src/main/java/org/openrewrite/java/testing/junit5/TimeoutRuleToClassAnnotation.java

```diff
@@ -0,0 +1,129 @@
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class TimeoutRuleToClassAnnotation extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "JUnit 4 `@Rule Timeout` to JUnit Jupiter's `Timeout`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace usages of JUnit 4's `@Rule Timeout` with JUnit 5 `Timeout` class annotation.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesType<>("org.junit.rules.Timeout", false), new TimeoutRuleToClassAnnotationVisitor());
+    }
+
+    public static class TimeoutRuleToClassAnnotationVisitor extends JavaIsoVisitor<ExecutionContext> {
+
+        private JavaParser.@Nullable Builder<?, ?> javaParser;
+
+        private JavaParser.Builder<?, ?> javaParser(ExecutionContext ctx) {
+            if (javaParser == null) {
+                javaParser = JavaParser.fromJavaVersion()
+                        .classpathFromResources(ctx, "junit-jupiter-api-5", "hamcrest-3");
+            }
+            return javaParser;
+
+        }
+
+        @Override
+        public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+            J.ClassDeclaration cd = super.visitClassDeclaration(classDecl, ctx);
+
+            AtomicReference<Expression> initializer = new AtomicReference<>();
+
+            cd = cd.withBody(cd.getBody().withStatements(ListUtils.map(cd.getBody().getStatements(), statement -> {
+                if (statement instanceof J.VariableDeclarations) {
+                    //noinspection ConstantConditions
+                    if (TypeUtils.isOfClassType(((J.VariableDeclarations) statement).getTypeExpression().getType(),
+                            "org.junit.rules.Timeout")) {
+                        List<J.VariableDeclarations.NamedVariable> variables = ((J.VariableDeclarations) statement).getVariables();
+                        if (!variables.isEmpty()) {
+                            initializer.set(variables.get(0).getInitializer());
+                        }
+                        return null;
+                    }
+                }
+                return statement;
+            })));
+
+            if (initializer.get() != null) {
+                cd = insertTimeoutAnnotation(initializer.get(), cd, ctx);
+                maybeRemoveImport("org.junit.Rule");
+                maybeRemoveImport("org.junit.rules.Timeout");
```

Comment:
Where possible I prefer a direct return as opposed to repeated assignments to the same variable.
```suggestion
                maybeRemoveImport("org.junit.Rule");
                maybeRemoveImport("org.junit.rules.Timeout");
                return insertTimeoutAnnotation(initializer.get(), cd, ctx);
```